src/master/production/sync-replication.md

```diff
@@ -192,21 +192,205 @@ Make sure you already have [ImmuDB installed](../running/download.md).
+   --replication-sync-enabled \
+   --replication-sync-followers 1 \
+   --replication-is-replica=false
+database 'replicadb' {replica: false} successfully updated
+```
+
+> Note that the number of required sync followers may be temporarily lowered due to the loss of the primary node.
+
+#### Step 3. Switch other replicas to follow new primary
+
+```shell
+$ immuadmin database update replicadb -p 3325 \
+   --replication-master-address 127.0.0.1 \
+   --replication-master-database replicadb \
+   --replication-master-port 3324
+```
+
+#### Step 4. Truncate precommitted transactions on other replicas if needed
+
+It may happen that the new replica will reject synchronizing with the new primary.
+In such case immudb will report an error in logs:
+
+```text
+immudb 2022/10/11 15:57:42 ERROR: follower commit state at 'replicadb' diverged from master's
+```
+
+To fix that issue please restart immudb with the `--replication-allow-tx-discarding` flag that will
+discard any transaction on the replica that has not yet been fully committed.
+
+#### Step 5. Start a new replica to restore original cluster size
+
+Because the primary node was irrecoverably lost, a new replica should be spawned in its place.
+Please refer to the previous section dealing with the loss of replica for more details
+on how to add a replacement replica node.
+
+#### Step 6. Point immudb clients to the new primary node
+
+Clients performing write operations should now be switched to the new primary node.
+
+</WrappedSection>
+
+<WrappedSection>
+
+## Changing configuration of a locked primary database
+
+In most cases the primary node can be easily updated and the change will be applied without the need for a restart.
+That way the primary node can change the number of required confirmations, turn on and of synchronous replication
+and even be changed to a replica.
+
+There can be a situation though where the database is already blocked with writes waiting for confirmations from
+replicas. This could happen if replicas became unavailable or as a result of misconfiguration where too high value of
+replica confirmations was set.
+
+In this situation trying to change the configuration of the database will lock as well and will be unblocked once
+the database itself continue committing transactions.
+
+If the database can not be fixed to restore commits (e.g. if it is impossible to add enough synced replicas quick enough),
+the following workaround can be used (please note that it requires immudb restart):
+
+1. Update database settings, e.g. run `immuadmin database update` command - that operation will lock indefinitely but will
+   already persist new database settings
+2. Restart the immudb database instance - upon restart, the configuration of the database is read and applied from persistent settings
+   thus it will apply the configuration set in previous step.
+
+With this approach, the number of required confirmations can be lowered down to correct value or even disabled
```

Comment:
to the correct value, or disabled to switch to asynchornous replication mode maybe?